provision-cluster/lib/utils.js

```diff
@@ -0,0 +1,24 @@
+const { v4: uuidv4 } = require('uuid');
+const fs = require('fs');
+
+function getUniqueClusterName() {
+  const repoName = process.env['GITHUB_REPOSITORY'].replace(/^.*\//, '');
+  const branch = process.env['GITHUB_HEAD_REF'];
+  const sha = process.env['GITHUB_SHA'].substring(0, 8);
+	const uuid = uuidv4().replace('-', '').substring(0, 8);
+
+  let name = `test-${uuid}-${repoName}-${sha}-${branch}`;
+  let sanitizedName = name.replace(/[^A-Za-z0-9-]/g, '-').replace(/-+$/g, '').toLowerCase().substring(0, 63);
+
+	return sanitizedName;
+}
```

Comment:
I like that you're including the repo and branch and sha. I did something simpler, just `test-<uid>`. I think when we integrate our work I can use this function. The one thing I'd suggest though is consider using the crypto module rather than a uuid in order to generate the random portion, something like this. The main reason I suggest that is that uuids in general are not uniformly random, so taking a substring of them (depending on the category) could have higher chances of collision. Using the crypto module as below will give you the exact number of bytes you want and ensure they are maximally random (for that number of bytes). Also the crypto module is built into nodejs, so it shrinks our dependency set to use it, although that's not a huge deal.

```
const crypto = require('crypto')

function uid() {
  return crypto.randomBytes(16).toString("hex")
}

```